### PR TITLE
Cleanup legacy Share.php from unused code

### PR DESCRIPTION
**Pull request by MorrisJobke**
The code was only used by tests and thus this reduces the code to the used paths and removes the branches that wouldn't be used anyway to reduce complexity and gain some overview while working towards completely removing this class.

Found most of this in my local stashes.

I mostly just reduced the branches by checking all the possible inputs and then eliminated the unused branches and methods.

### CODE REVIEW

**Comment by ChristophWurst** [comment]
> Found most of this in my local stashes.

I would appreciate if you submit such cleanups earlier. It feels a bit scary to remove so much code now :P

File: lib/private/Share/Share.php [hunk 1]
```diff
@@ -321,124 +323,33 @@ public static function getItemShared($itemType, $itemSource, $format = self::FOR
...
 		$backend = self::getBackend($itemType);
-		$l = \OC::$server->getL10N('lib');
 
 		if ($backend->isShareTypeAllowed($shareType) === false) {
-			$message = 'Sharing %s failed, because the backend does not allow shares from type %i';
-			$message_t = $l->t('Sharing %s failed, because the backend does not allow shares from type %i', array($itemSourceName, $shareType));
-			\OCP\Util::writeLog('OCP\Share', sprintf($message, $itemSourceName, $shareType), ILogger::DEBUG);
-			throw new \Exception($message_t);
+			$message = 'Sharing failed, because the backend does not allow shares from type %i';
+			throw new \Exception(sprintf($message, $shareType));
```
Comment (kesselb) [inline lib/private/Share/Share.php]: I think you can do this without sprintf

File: lib/private/Share/Share.php [hunk 2]
```diff
@@ -321,124 +323,33 @@ public static function getItemShared($itemType, $itemSource, $format = self::FOR
...
-				$inGroup = array_intersect($groupsOwner, $groupsShareWith);
-				if (empty($inGroup)) {
-					$message = 'Sharing %1$s failed, because the user '
-						.'%2$s is not a member of any groups that %3$s is a member of';
-					$message_t = $l->t('Sharing %1$s failed, because the user %2$s is not a member of any groups that %3$s is a member of', array($itemName, $shareWith, $uidOwner));
-					\OCP\Util::writeLog('OCP\Share', sprintf($message, $itemName, $shareWith, $uidOwner), ILogger::DEBUG);
-					throw new \Exception($message_t);
-				}
+				$message = 'Sharing failed, because the user %s does not exist';
+				throw new \Exception(sprintf($message, $shareWith));
```
Comment (kesselb) [inline lib/private/Share/Share.php]: No need for sprintf

File: lib/private/Share/Share.php [hunk 3]
```diff
@@ -447,10 +358,8 @@ public static function shareItem($itemType, $itemSource, $shareType, $shareWith,
 				// owner and is not a user share, this use case is for increasing
 				// permissions for a specific user
 				if ($checkExists['uid_owner'] != $uidOwner || $checkExists['share_type'] == $shareType) {
-					$message = 'Sharing %1$s failed, because this item is already shared with %2$s';
-					$message_t = $l->t('Sharing %1$s failed, because this item is already shared with %2$s', array($itemSourceName, $shareWith));
-					\OCP\Util::writeLog('OCP\Share', sprintf($message, $itemSourceName, $shareWith), ILogger::DEBUG);
-					throw new \Exception($message_t);
+					$message = 'Sharing failed, because this item is already shared with %s';
+					throw new \Exception(sprintf($message, $shareWith));
```
Comment (kesselb) [inline lib/private/Share/Share.php]: No need for sprintf

File: lib/private/Share/Share.php [hunk 4]
```diff
@@ -1711,10 +1327,7 @@ private static function checkReshare($itemType, $itemSource, $shareType, $shareW
 				}
 			} else {
 				$message = 'Sharing %s failed, because resharing is not allowed';
-				$message_t = $l->t('Sharing %s failed, because resharing is not allowed', array($itemSourceName));
-
-				\OCP\Util::writeLog('OCP\Share', sprintf($message, $itemSourceName), ILogger::DEBUG);
-				throw new \Exception($message_t);
+				throw new \Exception(sprintf($message, $itemSourceName));
```
Comment (kesselb) [inline lib/private/Share/Share.php]: No need for sprintf

File: lib/private/Share/Share.php [hunk 5]
```diff
@@ -459,188 +368,14 @@ public static function shareItem($itemType, $itemSource, $shareType, $shareWith,
...
-			if ($checkExists = self::getItems($itemType, $itemSource, self::SHARE_TYPE_GROUP, $shareWith,
-				null, self::FORMAT_NONE, null, 1, true, true)) {
-
-				if ($checkExists['share_with'] === $shareWith && $checkExists['share_type'] === \OCP\Share::SHARE_TYPE_GROUP) {
-					$message = 'Sharing %1$s failed, because this item is already shared with %2$s';
-					$message_t = $l->t('Sharing %1$s failed, because this item is already shared with %2$s', array($itemSourceName, $shareWith));
-					\OCP\Util::writeLog('OCP\Share', sprintf($message, $itemSourceName, $shareWith), ILogger::DEBUG);
-					throw new \Exception($message_t);
+					$message = 'Sharing failed, because this item is already shared with user %s';
+					throw new \Exception(sprintf($message, $shareWith));
```
Comment (kesselb) [inline lib/private/Share/Share.php]: No need for sprintf

**Review by kesselb — COMMENTED** [review]
I have no idea what you're doing here :see_no_evil:

File: lib/private/Share/Share.php (hunk 1 above)
Comment (MorrisJobke) [reply lib/private/Share/Share.php]: Just copied over the stuff from one line above ;)

**Comment by MorrisJobke** [comment]
> I would appreciate if you submit such cleanups earlier. It feels a bit scary to remove so much code now :P

Fine with moving this to NC 17. Was just laying around and thought it would be good to submit this as PR. 😉

**Comment by rullzer** [comment]
rebased.
Lets try to get this in

**Review by rullzer — APPROVED** [review]

**Comment by MorrisJobke** [comment]
@kesselb I would leave it as it is now, because I just copied it and it is planned to be removed anyways soon.

The only failing CI job is the Phan due to memory issues on the host (same on master). So this is ready for merge.

**Review by ChristophWurst — APPROVED** [review]
:elephant:

**Pull request merged**